src/Microsoft.AspNet.Server.Kestrel/Http/Listener.cs

```diff
@@ -13,8 +13,10 @@ namespace Microsoft.AspNet.Server.Kestrel.Http
     /// </summary>
     public abstract class Listener : ListenerContext, IDisposable
     {
-        protected Listener(ServiceContext serviceContext) : base(serviceContext)
+        protected Listener(ServiceContext serviceContext) 
+            : base(serviceContext)
         {
+            Memory2 = new MemoryPool2();
```

Comment:
So this replaces the IMemoryPool in general? They seem unrelated. InputPool and MemoryPool?